### PR TITLE
feat: upgrade protocol 2.4.0 + royalties management

### PR DESCRIPTION
**Pull request by levalleux-ludo**
## Description

{{what was done}}

## How to test

{{how can it be tested}}

### CODE REVIEW

File: packages/core-sdk/src/accounts/interface.ts [hunk 1]
```diff
@@ -273,3 +286,61 @@ export function decodeCalculateCollectionAddress(result: string): {
...
+
+export function decodeGetRoyaltyRecipients(
+  result: string
+): RoyaltyRecipientInfo[] {
+  const [royaltyRecipients] = bosonAccountHandlerIface.decodeFunctionResult(
+    "getRoyaltyRecipients",
+    result
+  );
+  return royaltyRecipients;
+  // // Temporary patch (until the protocol ABI for getRoyaltyRecipients() is fixed)
```
Comment (albertfolch-redeemeum) [inline packages/core-sdk/src/accounts/interface.ts:335]: remove or add "TODO:"

File: packages/subgraph/src/mappings/exchange-handler.ts [hunk 2]
```diff
@@ -29,6 +29,9 @@ export function handleBuyerCommittedEvent(event: BuyerCommitted): void {
   }
 
   const offer = Offer.load(exchangeFromEvent.offerId.toString());
+  log.debug("exchangeFromEvent.offerId.toString() {}", [
```
Comment (albertfolch-redeemeum) [inline packages/subgraph/src/mappings/exchange-handler.ts:32]: needed?

**Review by albertfolch-redeemeum — APPROVED** [review]
LGTM 🚀

File: packages/subgraph/src/mappings/exchange-handler.ts (hunk 2 above)
Comment (levalleux-ludo) [reply packages/subgraph/src/mappings/exchange-handler.ts]: cleaned code

**Pull request merged**